.github/workflows/pr_build.yaml

```
@@ -0,0 +1,102 @@
+name: Build Binaries (Pull Request)
+
+on:
+  pull_request:
+
+env:
+  BUILD_TYPE: RelWithDebInfo
+
+permissions:
+  contents: write
+
+jobs:
+  gen-build-version:
+    name: Generate Build Version
+    runs-on: ubuntu-latest
+    outputs:
+      build: ${{ steps.build_name.outputs.build }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Get the cargo package version
+        run: |
+          VERSION=$(awk -F '"' '/^version =/ { print $2 }' Cargo.toml)
+          echo "The package version is $VERSION"
+          echo "CARGO_PKG_VERSION=$VERSION" >> $GITHUB_ENV
+      - name: Set short git commit SHA
+        shell: bash
+        run: |
+          CALCULATED_SHA=$(git rev-parse --short ${{ github.sha }})
+          echo "SHA_SHORT=$CALCULATED_SHA" >> $GITHUB_ENV
+      - name: Confirm git commit SHA output
+        run: echo $SHA_SHORT
+      - name: Extract branch name
+        shell: bash
+        run: echo "BRANCH=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_ENV
+      - name: Confirm nightly build name
+        id: build_name
+        run: |
+          echo ${CARGO_PKG_VERSION}-${BRANCH}-${SHA_SHORT}
+          echo build=${CARGO_PKG_VERSION}-${BRANCH}-${SHA_SHORT} >> $GITHUB_OUTPUT
```

I think technically a branch can have spaces, so perhaps we should quote this:
```suggestion
          echo build="${CARGO_PKG_VERSION}-${BRANCH}-${SHA_SHORT}" >> $GITHUB_OUTPUT
```